amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/common/fm/FMSettingsTree.java

```diff
@@ -0,0 +1,41 @@
+package org.digijava.kernel.ampapi.endpoints.common.fm;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import io.swagger.annotations.ApiModelProperty;
+
+public class FMSettingsTree {
+    
+    @ApiModelProperty(value = "list of FM settings listed as a tree",
+            example = "Detail tree & fully enabled paths =>\n"
+                    + "    ```\n"
+                    + "    \"REPORTING\": {\n"
+                    + "         \"Measures\": {\n"
+                    + "             \"Actual Disbursements\": {},\n"
+                    + "             ...\n"
+                    + "    ```\n"
+                    + "\n"
+                    + "Detail tree & fully enabled paths =>\n"
+                    + "    ```\n"
+                    + "    \"REPORTING\": {\n"
+                    + "         \"__enabled\" : true, // omitted if fullEnabledPaths are requested (same below)\n"
+                    + "         \"Measures\": {\n"
+                    + "             \"__enabled\" : true,\n"
+                    + "             \"Actual Disbursements\": {\n"
+                    + "                 \"__enabled\" : true\n"
+                    + "             },\n"
+                    + "             ...\n"
+                    + "    ```\n")
+    private Map<String, Object> modules = new HashMap<>();
```

Comment:
Looking at structure I think you can do this:
```
public class FMSettingsTree {
   @JsonProperty("__enabled")
    boolean enabled;
   
   private Map<String, FMSettingsTree> tree; // still use JsonAnyGetter for it
}
```

Reply:
And use `@JsonUnwrapp` for `T fmSettings`;

Reply:
Great idea!
P.S.: we don't need to use @JsonUnwrapp

Reply:
True, no need for unwrapping, I forgot again we report "fm-settings" explicitly.